### PR TITLE
[WIP]: Issue #1141 Remove expects from amethyst_network

### PR DESCRIPTION
**Pull request by Plasticcaz**

### CODE REVIEW

**Comment by Plasticcaz** [comment]
I'll run rustfmt later to fix the above failure...

image
I am currently unsure how we should go about removing the above- expects found in network_socket.rs. The method doesn't return a result, so we can't offload responsibility for handling this error further up the chain. 

The first expect pictured there seems like an assertion that it should never fail, so I think we should leave it there.

I'm not very familiar with the nitty-gritty of networking code, but I was wondering if maybe I can downgrade the second expect to just log a warning if we reach it? It doesn't seem to be too serious, unless it would continually keep sending stop requests to no avail. Any suggestions?

File: amethyst_network/src/network_socket.rs [hunk 1]
```diff
@@ -89,9 +89,12 @@ where
         let mut socket = UdpSocket::bind(addr, NetworkConfig::default())
             .map_err(|x| Error::new(ErrorKind::Other, x.to_string()))?;
 
-        socket
-            .set_nonblocking(true)
-            .expect("Unable to set `UdpSocket` to non-blocking mode");
+        socket.set_nonblocking(true).map_err(|_| {
+            Error::new(
+                ErrorKind::Other,
+                "Unable to set `UdpSocket` to non-blocking mode".to_owned(),
```
Comment (udoprog) [inline amethyst_network/src/network_socket.rs]: `.to_owned()` is not necessary, all though it is worth noting that it still [does convert it into a String internally](https://doc.rust-lang.org/src/std/error.rs.html#271).

Comment (Plasticcaz) [reply amethyst_network/src/network_socket.rs]: Hmmm... I could have sworn it wasn't letting me pass in a string slice. Must have been from when I was trying something else. I'll remove it.

I don't think it matters so much about allocating the String, as it only happens on the non-happy path.

**Comment by TimonPost** [comment]
I come back at your question. I encountered the same indeed.

**Comment by azriel91** [comment]
Yo @Plasticcaz! do you expect to remove more expects in this PR, or are we good to merge?

**Comment by torkleyy** [comment]
I'd personally recommend not fixing too many of those at once, because that can become quite boring over time.

**Comment by TimonPost** [comment]
Let's merge this I'm fine with the expects there are currently.

**Comment by TimonPost** [comment]
bors r=TimonPost,LucioFranco,Xaeroxe,udoprog

**Comment by TimonPost** [comment]
bors r=TimonPost,LucioFranco,Xaeroxe,udoprog

**Comment by Plasticcaz** [comment]
> 
> 
> Yo @Plasticcaz! do you expect to remove more expects in this PR, or are we good to merge?

This issue was just to remove expects from networking code. As far as I could see there were 4 instances of expect. I removed two of them by returning error results instead. The other two didn't look so easy to get rid of, and I was waiting on feedback.

Looks like it's been decided to leave them for now. Thanks everyone!

**Comment by torkleyy** [comment]
Thank you @Plasticcaz!

**Pull request merged**